File: flutter_zoom_videosdk-2.4.0/ios/Classes/FlutterZoomVideoSdkSessionStatisticsInfo.m
#import "FlutterZoomVideoSdkSessionStatisticsInfo.h"
#import "JSONConvert.h"

@implementation FlutterZoomVideoSdkSessionStatisticsInfo

- (ZoomVideoSDKSession *)getSession {
    ZoomVideoSDKSession* session = nil;

    @try {
        session = [[ZoomVideoSDK shareInstance] getSession];
        if (session == nil) {
            NSException *e = [NSException exceptionWithName:@"NoSessionFound" reason:@"No Session Found" userInfo:nil];
            @throw e;
        }
    } @catch (NSException *e) {
        NSLog(@"%@ - %@", e.name, e.reason);
    }

    return session;
}

// -----------------------------------------------------------------------------------------------
// Audio Statistics Info
// -----------------------------------------------------------------------------------------------

-(void) getAudioStatisticsInfo:(FlutterResult) result {
    NSInteger recvFrequency = [[[self getSession] getSessionAudioStatisticInfo] recvFrequency];
    NSInteger recvJitter = [[[self getSession] getSessionAudioStatisticInfo] recvJitter];
    NSInteger recvLatency = [[[self getSession] getSessionAudioStatisticInfo] recvLatency];
    CGFloat recvPacketLossAvg = [[[self getSession] getSessionAudioStatisticInfo] recvPacketLossAvg];
    CGFloat recvPacketLossMax = [[[self getSession] getSessionAudioStatisticInfo] recvPacketLossMax];
    NSInteger sendFrequency = [[[self getSession] getSessionAudioStatisticInfo] sendFrequency];
    NSInteger sendJitter = [[[self getSession] getSessionAudioStatisticInfo] sendJitter];
    NSInteger sendLatency = [[[self getSession] getSessionAudioStatisticInfo] sendLatency];
    CGFloat sendPacketLossAvg = [[[self getSession] getSessionAudioStatisticInfo] sendPacketLossAvg];
    CGFloat sendPacketLossMax = [[[self getSession] getSessionAudioStatisticInfo] sendPacketLossMax];
    NSDictionary *dict = @{
        @"recvFrequency": @(recvFrequency),
        @"recvJitter": @(recvJitter),
        @"recvLatency": @(recvLatency),
        @"recvPacketLossAvg": @(recvPacketLossAvg),
        @"recvPacketLossMax": @(recvPacketLossMax),
        @"sendFrequency": @(sendFrequency),
        @"sendJitter": @(sendJitter),
        @"sendLatency": @(sendLatency),
        @"sendPacketLossAvg": @(sendPacketLossAvg),
        @"sendPacketLossMax": @(sendPacketLossMax),
    };

    result([JSONConvert NSDictionaryToNSString: dict]);
}

// -----------------------------------------------------------------------------------------------
// Video Statistics Info
// -----------------------------------------------------------------------------------------------

-(void) getVideoStatisticsInfo:(FlutterResult) result {
    NSInteger recvFps = [[[self getSession] getSessionVideoStatisticInfo] recvFps];
    NSInteger recvFrameHeight = [[[self getSession] getSessionVideoStatisticInfo] recvFrameHeight];
    NSInteger recvFrameWidth = [[[self getSession] getSessionVideoStatisticInfo] recvFrameWidth];
    NSInteger recvJitter = [[[self getSession] getSessionVideoStatisticInfo] recvJitter];
    NSInteger recvLatency = [[[self getSession] getSessionVideoStatisticInfo] recvLatency];
    CGFloat recvPacketLossAvg = [[[self getSession] getSessionVideoStatisticInfo] recvPacketLossAvg];
    CGFloat recvPacketLossMax = [[[self getSession] getSessionVideoStatisticInfo] recvPacketLossMax];
    NSInteger sendFps = [[[self getSession] getSessionVideoStatisticInfo] sendFps];
    NSInteger sendFrameHeight = [[[self getSession] getSessionVideoStatisticInfo] sendFrameHeight];
    NSInteger sendFrameWidth = [[[self getSession] getSessionVideoStatisticInfo] sendFrameWidth];
    NSInteger sendJitter = [[[self getSession] getSessionVideoStatisticInfo] sendJitter];
    NSInteger sendLatency = [[[self getSession] getSessionVideoStatisticInfo] sendLatency];
    CGFloat sendPacketLossAvg = [[[self getSession] getSessionVideoStatisticInfo] sendPacketLossAvg];
    CGFloat sendPacketLossMax = [[[self getSession] getSessionVideoStatisticInfo] sendPacketLossMax];
    NSDictionary *dict = @{
        @"recvFps": @(recvFps),
        @"recvFrameHeight": @(recvFrameHeight),
        @"recvFrameWidth": @(recvFrameWidth),
        @"recvJitter": @(recvJitter),
        @"recvLatency": @(recvLatency),
        @"recvPacketLossAvg": @(recvPacketLossAvg),
        @"recvPacketLossMax": @(recvPacketLossMax),
        @"sendFps": @(sendFps),
        @"sendFrameHeight": @(sendFrameHeight),
        @"sendFrameWidth": @(sendFrameWidth),
        @"sendJitter": @(sendJitter),
        @"sendLatency": @(sendLatency),
        @"sendPacketLossAvg": @(sendPacketLossAvg),
        @"sendPacketLossMax": @(sendPacketLossMax),
    };

    result([JSONConvert NSDictionaryToNSString: dict]);
}

// -----------------------------------------------------------------------------------------------
// Share Statistics Info
// -----------------------------------------------------------------------------------------------

-(void) getShareStatisticsInfo:(FlutterResult) result {
    NSInteger recvFps = [[[self getSession] getSessionShareStatisticInfo] recvFps];
    NSInteger recvFrameHeight = [[[self getSession] getSessionShareStatisticInfo] recvFrameHeight];
    NSInteger recvFrameWidth = [[[self getSession] getSessionShareStatisticInfo] recvFrameWidth];
    NSInteger recvJitter = [[[self getSession] getSessionShareStatisticInfo] recvJitter];
    NSInteger recvLatency = [[[self getSession] getSessionShareStatisticInfo] recvLatency];
    CGFloat recvPacketLossAvg = [[[self getSession] getSessionShareStatisticInfo] recvPacketLossAvg];
    CGFloat recvPacketLossMax = [[[self getSession] getSessionShareStatisticInfo] recvPacketLossMax];
    NSInteger sendFps = [[[self getSession] getSessionShareStatisticInfo] sendFps];
    NSInteger sendFrameHeight = [[[self getSession] getSessionShareStatisticInfo] sendFrameHeight];
    NSInteger sendFrameWidth = [[[self getSession] getSessionShareStatisticInfo] sendFrameWidth];
    NSInteger sendJitter = [[[self getSession] getSessionShareStatisticInfo] sendJitter];
    NSInteger sendLatency = [[[self getSession] getSessionShareStatisticInfo] sendLatency];
    CGFloat sendPacketLossAvg = [[[self getSession] getSessionShareStatisticInfo] sendPacketLossAvg];
    CGFloat sendPacketLossMax = [[[self getSession] getSessionShareStatisticInfo] sendPacketLossMax];
    NSDictionary *dict = @{
        @"recvFps": @(recvFps),
        @"recvFrameHeight": @(recvFrameHeight),
        @"recvFrameWidth": @(recvFrameWidth),
        @"recvJitter": @(recvJitter),
        @"recvLatency": @(recvLatency),
        @"recvPacketLossAvg": @(recvPacketLossAvg),
        @"recvPacketLossMax": @(recvPacketLossMax),
        @"sendFps": @(sendFps),
        @"sendFrameHeight": @(sendFrameHeight),
        @"sendFrameWidth": @(sendFrameWidth),
        @"sendJitter": @(sendJitter),
        @"sendLatency": @(sendLatency),
        @"sendPacketLossAvg": @(sendPacketLossAvg),
        @"sendPacketLossMax": @(sendPacketLossMax),
    };

    result([JSONConvert NSDictionaryToNSString: dict]);
}

@end

File: flutter_zoom_videosdk-2.4.0/android/src/main/java/com/flutterzoom/videosdk/FlutterZoomVideoSdkSessionStatisticsInfo.java
package com.flutterzoom.videosdk;

import androidx.annotation.NonNull;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import java.util.HashMap;
import java.util.Map;

import io.flutter.plugin.common.MethodChannel;
import us.zoom.sdk.ZoomVideoSDK;
import us.zoom.sdk.ZoomVideoSDKSession;

public class FlutterZoomVideoSdkSessionStatisticsInfo {

    private ZoomVideoSDKSession getSession() {
        ZoomVideoSDKSession session = null;
        try {
            session = ZoomVideoSDK.getInstance().getSession();
            if (session == null) {
                throw new Exception("No Session Found");
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return session;
    }

    // -----------------------------------------------------------------------------------------------
    // region Audio Statistics Info
    // -----------------------------------------------------------------------------------------------
    
    public void getAudioStatisticsInfo(@NonNull MethodChannel.Result result) {
        int recvFrequency = getSession().getSessionAudioStatisticInfo().getRecvFrequency();
        int recvJitter = getSession().getSessionAudioStatisticInfo().getRecvJitter();
        int recvLatency = getSession().getSessionAudioStatisticInfo().getRecvLatency();
        float recvPacketLossAvg = getSession().getSessionAudioStatisticInfo().getRecvPacketLossAvg();
        float recvPacketLossMax = getSession().getSessionAudioStatisticInfo().getRecvPacketLossMax();
        int sendFrequency = getSession().getSessionAudioStatisticInfo().getSendFrequency();
        int sendJitter = getSession().getSessionAudioStatisticInfo().getSendJitter();
        int sendLatency = getSession().getSessionAudioStatisticInfo().getSendLatency();
        float sendPacketLossAvg = getSession().getSessionAudioStatisticInfo().getSendPacketLossAvg();
        float sendPacketLossMax = getSession().getSessionAudioStatisticInfo().getSendPacketLossMax();

        Map<String, Object> info = new HashMap<>();
        info.put("recvFrequency", recvFrequency);
        info.put("recvJitter", recvJitter);
        info.put("recvLatency", recvLatency);
        info.put("recvPacketLossAvg", recvPacketLossAvg);
        info.put("recvPacketLossMax", recvPacketLossMax);
        info.put("sendFrequency", sendFrequency);
        info.put("sendJitter", sendJitter);
        info.put("sendLatency", sendLatency);
        info.put("sendPacketLossAvg", sendPacketLossAvg);
        info.put("sendPacketLossMax", sendPacketLossMax);

        Gson gson = new GsonBuilder().create();
        String jsonAudioStatisticsInfo = gson.toJson(info);


        result.success(jsonAudioStatisticsInfo);
    }

    // -----------------------------------------------------------------------------------------------
    // endregion
    // -----------------------------------------------------------------------------------------------

    // -----------------------------------------------------------------------------------------------
    // region Video Statistics Info
    // -----------------------------------------------------------------------------------------------
    
    public void getVideoStatisticsInfo(@NonNull MethodChannel.Result result) {
        int recvFps = getSession().getSessionVideoStatisticInfo().getRecvFps();
        int recvFrameHeight = getSession().getSessionVideoStatisticInfo().getRecvFrameHeight();
        int recvFrameWidth = getSession().getSessionVideoStatisticInfo().getRecvFrameWidth();
        int recvJitter = getSession().getSessionVideoStatisticInfo().getRecvJitter();
        int recvLatency = getSession().getSessionVideoStatisticInfo().getRecvLatency();
        float recvPacketLossAvg = getSession().getSessionVideoStatisticInfo().getRecvPacketLossAvg();
        float recvPacketLossMax = getSession().getSessionVideoStatisticInfo().getRecvPacketLossMax();
        int sendFps = getSession().getSessionVideoStatisticInfo().getSendFps();
        int sendFrameHeight = getSession().getSessionVideoStatisticInfo().getSendFrameHeight();
        int sendFrameWidth = getSession().getSessionVideoStatisticInfo().getSendFrameWidth();
        int sendJitter = getSession().getSessionVideoStatisticInfo().getSendJitter();
        int sendLatency = getSession().getSessionVideoStatisticInfo().getSendLatency();
        float sendPacketLossAvg = getSession().getSessionVideoStatisticInfo().getSendPacketLossAvg();
        float sendPacketLossMax = getSession().getSessionVideoStatisticInfo().getSendPacketLossMax();

        Map<String, Object> info = new HashMap<>();
        info.put("recvFps", recvFps);
        info.put("recvFrameHeight", recvFrameHeight);
        info.put("recvFrameWidth", recvFrameWidth);
        info.put("recvJitter", recvJitter);
        info.put("recvLatency", recvLatency);
        info.put("recvPacketLossAvg", recvPacketLossAvg);
        info.put("recvPacketLossMax", recvPacketLossMax);
        info.put("sendFps", sendFps);
        info.put("sendFrameHeight", sendFrameHeight);
        info.put("sendFrameWidth", sendFrameWidth);
        info.put("sendJitter", sendJitter);
        info.put("sendLatency", sendLatency);
        info.put("sendPacketLossAvg", sendPacketLossAvg);
        info.put("sendPacketLossMax", sendPacketLossMax);

        Gson gson = new GsonBuilder().create();
        String jsonVideoStatisticsInfo = gson.toJson(info);


        result.success(jsonVideoStatisticsInfo);
    }

    // -----------------------------------------------------------------------------------------------
    // endregion
    // -----------------------------------------------------------------------------------------------

    // -----------------------------------------------------------------------------------------------
    // region Share Statistics Info
    // -----------------------------------------------------------------------------------------------

    
    public void getShareStatisticsInfo(@NonNull MethodChannel.Result result) {
        int recvFps = getSession().getSessionShareStatisticInfo().getRecvFps();
        int recvFrameHeight = getSession().getSessionShareStatisticInfo().getRecvFrameHeight();
        int recvFrameWidth = getSession().getSessionShareStatisticInfo().getRecvFrameWidth();
        int recvJitter = getSession().getSessionShareStatisticInfo().getRecvJitter();
        int recvLatency = getSession().getSessionShareStatisticInfo().getRecvLatency();
        float recvPacketLossAvg = getSession().getSessionShareStatisticInfo().getRecvPacketLossAvg();
        float recvPacketLossMax = getSession().getSessionShareStatisticInfo().getRecvPacketLossMax();
        int sendFps = getSession().getSessionShareStatisticInfo().getSendFps();
        int sendFrameHeight = getSession().getSessionShareStatisticInfo().getSendFrameHeight();
        int sendFrameWidth = getSession().getSessionShareStatisticInfo().getSendFrameWidth();
        int sendJitter = getSession().getSessionShareStatisticInfo().getSendJitter();
        int sendLatency = getSession().getSessionShareStatisticInfo().getSendLatency();
        float sendPacketLossAvg = getSession().getSessionShareStatisticInfo().getSendPacketLossAvg();
        float sendPacketLossMax = getSession().getSessionShareStatisticInfo().getSendPacketLossMax();

        Map<String, Object> info = new HashMap<>();
        info.put("recvFps", recvFps);
        info.put("recvFrameHeight", recvFrameHeight);
        info.put("recvFrameWidth", recvFrameWidth);
        info.put("recvJitter", recvJitter);
        info.put("recvLatency", recvLatency);
        info.put("recvPacketLossAvg", recvPacketLossAvg);
        info.put("recvPacketLossMax", recvPacketLossMax);
        info.put("sendFps", sendFps);
        info.put("sendFrameHeight", sendFrameHeight);
        info.put("sendFrameWidth", sendFrameWidth);
        info.put("sendJitter", sendJitter);
        info.put("sendLatency", sendLatency);
        info.put("sendPacketLossAvg", sendPacketLossAvg);
        info.put("sendPacketLossMax", sendPacketLossMax);

        Gson gson = new GsonBuilder().create();
        String jsonShareStatisticsInfo = gson.toJson(info);


        result.success(jsonShareStatisticsInfo);
    }

    // -----------------------------------------------------------------------------------------------
    // endregion
    // -----------------------------------------------------------------------------------------------

}
